gensim/models/tfidfmodel.py

```diff
@@ -387,8 +419,36 @@ def __getitem__(self, bow, eps=1e-12):
 
         # and finally, normalize the vector either to unit length, or use a
         # user-defined normalization function
-        vector = self.normalize(vector)
+        if self.pivot_norm is False:
+            norm_vector = self.normalize(vector)
+            norm_vector = [(termid, weight) for termid, weight in norm_vector if abs(weight) > self.eps]
+            return norm_vector
+        else:
+            logger.info("You need to explicitly call pivoted_normalization.")
+            return vector
+
+    def pivoted_normalization(self, tfidf_matrix):
```

Comment:
:warning: This is the most suspicious place, need to investigate, how this can be done in different way :warning: